Makefile

```diff
@@ -210,6 +210,18 @@ build ${PROTOC_GEN_SWIFT}:
 	@rm Tests/LinuxMain.swift.new
 	${SWIFT} build
 
+# This will get run by any other rule that tries to use the plugin.  This hacks
+# in a check during the library build to ensure that the protoc on the local
+# system is one we expect. This was inspired by the findings that lead to
+#   https://github.com/apple/swift-protobuf/issues/111
+# This likely will need follow up to eventually do a pattern match to ensure it
+# is 3.1 or higher.
+${PROTOC_GEN_SWIFT}: build
+	@if [ "$(shell ${PROTOC} --version)" != "libprotoc 3.1.0" ]; then \
```

Comment:
Not sure.  There were a few months between the 3.0.0 release and 3.1.0 releases.  There were some 3.0.x releases, but that number didn't always make it to the master branch.

The other catch, is I'm think the only shell we can rely on make spawning in /bin/sh, and test (`[`) doesn't have a regex matcher either.  So when this needs beefing up, it make get ugly and be yet another reason to move from a Makefile to a script for wrapping the build/test/regenerate process.